Dockerfile

```diff
@@ -26,6 +26,10 @@ RUN apt-get dist-upgrade -y
 RUN wget -q -O- https://deb.nodesource.com/setup_10.x | bash -
 RUN apt-get install -y nodejs
 
+# Install youtube-dl
```

Comment:
For youtube-dl functionality, we can just add one of the many go implementations and have the server call it, instead of using a separate program.
Of course, any previous problems will not be an issue with this, since the video itself will be the only thing distributed.
Here's one that looks good, and even supports multiple sites other than youtube.
https://github.com/iawia002/annie

Reply:
Well this one is too released as a binary, so using it as a library is not reliable since functions may be changed.
youtube-dl has large community behind it with couple releases every month which is important due to frequent youtube changes (I've experineced couple of times that few months old release won't work) so using small project may cause problems in future.

Reply:
Again, this is also frequently updated, and it's not like the server is constantly getting the newest version of youtube-dl every day either.